docs/samples/s3/README.md

```diff
@@ -0,0 +1,117 @@
+
+# Predict on a KFService with saved model on S3
+## Setup
+1. Your ~/.kube/config should point to a cluster with [KFServing installed](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#deploy-kfserving).
+2. Your cluster's Istio Ingress gateway must be network accessible.
+3. Your cluster's Istio Egresss gateway must [allow accessing S3 Storage](https://knative.dev/docs/serving/outbound-network-access/)
+4. The example uses the Kubeflow's Minio setup if you have [Kubeflow](https://www.kubeflow.org/docs/started/getting-started/) installed,
+you can also setup your own [Minio server](https://docs.min.io/docs/deploy-minio-on-kubernetes.html) or use other S3 compatible cloud storage.
+
+
+## Train TF mnist model and save on S3
+Follow Kubeflow's [TF mnist example](https://github.com/kubeflow/examples/tree/master/mnist#using-s3) to train a TF mnist model and save on S3,
+change following S3 access settings, `modelDir` and `exportDir` as needed. If you already have a mnist model saved on S3 you can skip this step.
+```bash
+export S3_USE_HTTPS=0 #set to 0 for default minio installs
+export S3_ENDPOINT=minio-service.kubeflow:9000
+export AWS_ENDPOINT_URL=http://${S3_ENDPOINT}
+
+kustomize edit add configmap mnist-map-training --from-literal=S3_ENDPOINT=${S3_ENDPOINT}
+kustomize edit add configmap mnist-map-training --from-literal=AWS_ENDPOINT_URL=${AWS_ENDPOINT_URL}
+kustomize edit add configmap mnist-map-training --from-literal=S3_USE_HTTPS=${S3_USE_HTTPS}
+
+kustomize edit add configmap mnist-map-training --from-literal=modelDir=s3://mnist/v1
+kustomize edit add configmap mnist-map-training --from-literal=exportDir=s3://mnist/v1/export
+```
```

Comment:
We need to simplify the user experience. Assuming I as a user have specified the two yamls, I should not be required to do anything with kustomize?

Reply:
Agreed.

Reply:
these settings are actually for the kubeflow TF training job to save the model,  are you saying we directly provide the TF job yaml ? This step is not required if you already have a model on S3.

Reply:
I think I can just add the actual TF job yaml without above steps, but trust me the yaml looks a bit scary :)

Reply:
If I have not saved a model on s3, the steps would be more like
1. Use s3 or minio client to save your model on s3
2. Add your s3 location in yaml
3. Apply yaml.

Then we have automation to take the yaml and do what is needed in background?

Why the user is being asked to manipulate configmap settings? Maybe this is point in time, but this needs to be improved?

Reply:
Chatted with Animesh offline, this is the current way how kubeflow does for the TF training job, we can change to use arena cli for better user experience.